display/buffer.go

```diff
@@ -39,8 +42,16 @@ func DrawBuffer(screen tcell.Screen, palette *Palette, buffer *state.BufferState
 		} else if err != nil {
 			log.Fatalf("%s", err)
 		}
+
 		lineNum := textTree.LineNumForPosition(pos)
 		lineStartPos := textTree.LineStartPosition(lineNum)
+		if lineNumberMode == config.LineNumberModeRelative {
+			if lineNum < cursorLine {
+				lineNum = cursorLine - lineNum - 1
+			} else {
+				lineNum = lineNum - cursorLine - 1
+			}
+		}
```

Comment:
there's a small edge case here: in an empty document, the loop starting on line 38 gets skipped, and the line number gets hard-coded to `0` on line 81.

```go
	// Text view is empty, with cursor positioned in the first cell.
	if pos-viewTextOrigin == 0 && pos == cursorPos {
		sr.ShowCursor(int(lineNumMargin), 0)
		drawLineNumIfNecessary(sr, palette, 0, 0, lineNumMargin)
	}
```

which then gets displayed as `1` since `drawLineNumIfNecessary` adds one to the line number on line 196:
```go
lineNumStr := strconv.FormatUint(lineNum+1, 10)
```

I wonder if it might be easier to move the "line number to display based on lineNumMode" calculation into `drawLineNumIfNecessary` to cover this edge case?